src/level_four/core/controller/initial_device_provision_contoller.c

```diff
@@ -444,26 +446,29 @@ void lock_all_slots(){
 
     atecc_data.retries = DEFAULT_ATECC_RETRIES;
     bool lock = false;
+    uint32_t err_count=0;
     do
     {
-        if(atecc_data.status != ATCA_SUCCESS){
+        if(atecc_data.status != ATCA_SUCCESS || err_count != 0){
             LOG_CRITICAL("PERR4-0x%02x, retry-%d", atecc_data.status, atecc_data.retries);
         }
         for (uint32_t i = 0; i < sizeof(atecc_slot_to_lock); i++)
         {
             atecc_data.status = atcab_is_slot_locked(atecc_slot_to_lock[i], &lock);
             if(atecc_data.status != ATCA_SUCCESS){
-                break;
+                LOG_CRITICAL("PERR5=0x%02x, err=%d", atecc_slot_to_lock[i], atecc_data.status);
+                err_count++;
             }
             else if(lock == true){
                 continue;
             }
             atecc_data.status = atcab_lock_data_slot(atecc_slot_to_lock[i]);
             if(atecc_data.status != ATCA_SUCCESS){
-                break;
+                LOG_CRITICAL("PERR6=0x%02x, err=%d", atecc_slot_to_lock[i], atecc_data.status);
+                err_count++;;
             }
         }
-    } while (atecc_data.status != ATCA_SUCCESS && --atecc_data.retries);
+    } while ((atecc_data.status != ATCA_SUCCESS || err_count != 0) && --atecc_data.retries);
```

Comment:
In this condition, if `err_count` is non-zero but for that iteration all slots were locked successfully, do-while loop will run indefinitely as `err_count` is not reset to 0 before the iteration.

Reply:
err_count variable is reset in while loop